### ISSUE TITLE
Unhelpful parse error "Cannot return a forall quantifier"

### ISSUE THREAD

**Issue by ysangkok** [idris-lang/Idris2#2940]
# Steps to Reproduce
```
module Micro

f : forall a b. (Unit -> Unit) -> Unit -> Unit
f = id
```

# Expected Behavior
Compiles without error.

# Observed Behavior
```
 % idris2 -c Micro.idr 
1/1: Building Micro (Micro.idr)
Error: Cannot return a forall quantifier.

Micro:3:5--3:13
 1 | module Micro
 2 | 
 3 | f : forall a b. (Unit -> Unit) -> Unit -> Unit
         ^^^^^^^^
```

I don't understand how the forall is returned, since none of the type variables in the forall are mentioned the the right hand side of the last arrow.

Furthermore, if I remove the `b`, it compiles. If the `a` is the problem, as the highlighted part shows, how can removing the `b` fix the issue?

``` 
 % idris2 --version
Idris 2, version 0.6.0-b185f1ff8
```

**Comment by buzden** [comment]
Syntactically (unlike Haskell) there must be a comma between `a` and `b` in your signature. So, your code is not correct and should not compile. But the error message is indeed strange.

**Comment by dunhamsteve** [comment]
I believe this is addressed by #2941, but I forgot to put the magic word "fixes" in front of the bug number.

**Comment by ysangkok** [comment]
Excellent! Thank you so much!
